src/cart/services/cart.service.ts

```diff
@@ -34,18 +71,25 @@ export class CartService {
     return this.createByUserId(userId);
   }
 
-  updateByUserId(userId: string, { items }: Cart): Cart {
-    const { id, ...rest } = this.findOrCreateByUserId(userId);
-
-    const updatedCart = {
-      id,
-      ...rest,
-      items: [ ...items ],
+  async updateByUserId(userId: string, cart: Cart): Promise<Cart> {
+    const client = new Client(pgDbOptions);
+    await client.connect();
+    try {
+      for await (const cartItem of cart.items){
+        await client.query(`
```

Comment:
Please reconsider your approach to commands creation. Currently your code is vulnerable to SQL Injection attack.
Please use parameterized commands:
https://node-postgres.com/features/queries#parameterized-query